framework/test/unit/modules/dpos_v2/commands/unlock.spec.ts

```diff
@@ -430,4 +497,66 @@ describe('UnlockCommand', () => {
 			);
 		});
 	});
+
+	describe(`when certificate is not generated`, () => {
+		beforeEach(async () => {
+			await genesisSubstore.setWithSchema(
+				EMPTY_KEY,
+				{
+					height: 10,
+					initRounds: 1,
+					initDelegates: [],
+				},
+				genesisDataStoreSchema,
+			);
+			await delegateSubstore.setWithSchema(
+				delegate1.address,
+				{
+					name: delegate1.name,
+					...defaultDelegateInfo,
+				},
+				delegateStoreSchema,
+			);
+			await delegateSubstore.setWithSchema(
+				delegate2.address,
+				{
+					name: delegate2.name,
+					...defaultDelegateInfo,
+				},
+				delegateStoreSchema,
+			);
+			nonUnlockableObject = {
+				delegateAddress: delegate2.address,
+				amount: delegate2.amount,
+				unvoteHeight: blockHeight,
```

Comment:
if blockHeight is `unvoteHeight`, regardless of certificate, it will be unlockable?

Reply:
According to the logic: 
`for each unlockObject in voterStore(senderAddress).pendingUnlocks:
    if (hasWaited(unlockObject, senderAddress, height) 
        and not isPunished(unlockObject, senderAddress, height)
        and isCertificateGenerated(unlockObject)):
        delete unlockObject from voterStore(senderAddress).pendingUnlocks
        token.unlock(senderAddress, MODULE_ID_DPOS, TOKEN_ID_DPOS, unlockObject.amount)`